packages/server-wallet/src/protocols/process-ledger-queue.ts

```diff
@@ -0,0 +1,198 @@
+import {compose, map, filter} from 'lodash/fp';
+import _ from 'lodash';
+import {
+  allocateToTarget,
+  AllocationItem,
+  BN,
+  checkThat,
+  isSimpleAllocation,
+  Outcome,
+  SignedStateWithHash,
+  SimpleAllocation,
+} from '@statechannels/wallet-core';
+import {Transaction} from 'knex';
+
+import {Store} from '../wallet/store';
+import {Bytes32} from '../type-aliases';
+import {LedgerRequestType} from '../models/ledger-requests';
+
+import {Protocol, ProtocolResult, ChannelState} from './state';
+import {
+  MarkLedgerFundingRequestsAsComplete,
+  noAction,
+  ProtocolAction,
+  SignState,
+  signState,
+} from './actions';
+
+export type ProtocolState = {
+  fundingChannel: ChannelState;
+  channelsPendingRequest: ChannelState[];
+};
+
+const allocationItemComparator = (a: AllocationItem, b: AllocationItem): boolean =>
+  a.destination === b.destination && BN.eq(a.amount, b.amount);
+
+const foldAllocateToTarget = (
+  original: SimpleAllocation,
+  allocationTargets: ChannelState[]
+): SimpleAllocation =>
+  // This could fail at some point if there is no longer space to fund stuff
+  // TODO: Handle that case
+  // } catch (e) {
+  //   if (e.toString() === 'Insufficient funds in fundingChannel channel')
+  // }
+  allocationTargets.reduce(
+    (outcome, {channelId, supported}) =>
+      checkThat(
+        allocateToTarget(
+          outcome,
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          checkThat(supported!.outcome, isSimpleAllocation).allocationItems,
+          channelId
+        ),
+        isSimpleAllocation
+      ),
+    original
```

Comment:
💡 If you pass in a more strictly-typed `original` then you don't need as many `checkThat`s.

💡 💡 💡 If `ChannelState` were actually specialized to application channels (`SimpleAppChannelState`), the outcome on it could be typed as `SimpleAllocation`.